core/src/main/java/com/spotify/metrics/core/SemanticMetricRegistry.java

```diff
@@ -245,15 +254,15 @@ public void removeMatching(final SemanticMetricFilter filter) {
     }
 
     /**
-     * Adds a {@link SemanticMetricRegistryListener} to a collection of listeners that will be
+     * Adds a {@link SemanticMetricRegistryListenerV2} to a collection of listeners that will be
      * notified on
      * metric creation.  Listeners will be notified in the order in which they are added.
      * <p/>
      * <b>N.B.:</b> The listener will be notified of all existing metrics when it first registers.
      *
      * @param listener the listener that will be notified
      */
-    public void addListener(final SemanticMetricRegistryListener listener) {
+    public void addListener(final SemanticMetricRegistryListenerV2 listener) {
```

Comment:
Isn't this a breaking change? If someone depends on this library and calls `addListener( SemanticMetricRegistryListener)` they would now get MethodNotFound

Reply:
No, I don't think so. SemanticMetricRegistryListenerV2 is an extension of SemanticMetricRegistryListener. I don't see why you will get methodNotFound

Reply:
if the method signature is `void addListener(SemanticMetricRegistryListenerV2)` I can't pass a super-type of SemanticMetricRegistryListenerV2 to the method, I can only pass implementations of that type

for example, if you have a method `void foobar(String)` I can't pass an `Object` or `CharSequence` to that method

Reply:
Good catch, I had SemanticMetricRegistryListener in mind.